lib/bolt/node/winrm.rb

```diff
@@ -272,9 +272,10 @@ def _run_command(command)
     def _run_script(script, arguments)
       @logger.info { "Running script '#{script}'" }
       with_remote_file(script) do |remote_path|
-        args = [*PS_ARGS, '-File', "\"#{remote_path}\""]
+        path, args = *process_from_extension(remote_path)
+        # TODO: why do we escape in script, but not in task?
         args += escape_arguments(arguments)
-        execute_process('powershell.exe', args)
+        execute_process(path, args, stdin)
```

Comment:
This will require mocking a test given there is no Puppet / Ruby / etc installed on the remote host by default.